### DISCUSSION TITLE
How can I unit test a custom plugin?

### DISCUSSION THREAD

**Discussion by Vivi-Ornitier** [nuxt/nuxt#16973] [Questions]
I’m creating the following custom plugin.

```
export default defineNuxtPlugin(() => {
  return {
    provide: {
      hello: (msg: string) => `Hello ${msg}!`
    }
  }
})
```

I can use it in the template as follows.

```
<template>
  <div>
    {{ $hello('world') }}
  </div>
</template>
```

How can I unit test a custom plugin?

**Comment by mirumirumi** [comment]
Hi!
I think there are two patterns I see.

1. Normal function tests in custom plug-ins (return value assertions)
2. Verify the match against the expected display on the template side (i.e., similar to a test on the component side used)

What do you think? :)

**Reply by Vivi-Ornitier** [reply]
@mirumirumi 

Thank you for your reply!
How do I call the plugin in my test?

**Comment by JuanxCursed** [comment]
Hey guys, I'm trying to run the test for a few plugins/composables but I'm having issues with useNuxtApp, looks like that need to be mocked, but that doesn't make sense to me, can you guys provide some example of how did you achieve it?
